python/tchannel/tornado/connection.py

```diff
@@ -554,4 +574,19 @@ def send_request(self, request):
         future = tornado.gen.Future()
         self._outstanding[request.id] = future
         self.post_request(request)
+
+        def adapt_tracing(f):
+            # fetch the request tracing for response
+            f.result().tracing = request.tracing
```

Comment:
First if someone uses multi consumers for one future, I don't think they are using future in right way. Second let's step back. If one of the consumers mutates the future, then the second one may not get the same object regardless my change for the tracing.